src/appleseed/renderer/modeling/postprocessingstage/multithreadpostprocessingstage.h

```diff
@@ -0,0 +1,59 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2020 Tiago Chaves, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#pragma once
+
+// appleseed.renderer headers.
+#include "renderer/modeling/postprocessingstage/postprocessingstage.h"
+#include "renderer/modeling/postprocessingstage/effect/ipostprocessingeffect.h"
+
+// Forward declarations.
+namespace renderer  { class Frame; }
+namespace renderer  { class ParamArray; }
+
+namespace renderer
+{
+
+class MultithreadPostProcessingStage
+  : public PostProcessingStage
+{
+  public:
+    // Constructor.
+    MultithreadPostProcessingStage(
+        const char*             name,
+        const ParamArray&       params);
+
+    // Execute this post-processing stage on a given frame
+    // by scheduling a job for each tile to apply the image effect.
+    virtual void execute_on_tiles(
```

Comment:
Why aren't you overriding `PostProcessingStage::execute()` here? That was the idea: a multithreaded post-processing stage is just another "standard" stage, it just happens to execute a post-processing effect in a multithreaded fashion. The post-processing effect should be passed to the constructor.